component/src/main/java/io/siddhi/extension/io/s3/sink/S3Sink.java

```diff
@@ -0,0 +1,338 @@
+        parameters = {
+                @Parameter(
+                        name = "credential.provider.class",
+                        type = DataType.STRING,
+                        description = "AWS credential provider class to be used. If blank along with the username " +
+                                "and the password, default credential provider will be used.",
+                        optional = true,
+                        defaultValue = " "
+                ),
+                @Parameter(
+                        name = "aws.access.key",
+                        type = DataType.STRING,
+                        description = "AWS access key. This cannot be used along with the credential.provider.class",
+                        optional = true,
+                        defaultValue = " "
+                ),
+                @Parameter(
+                        name = "aws.secret.key",
+                        type = DataType.STRING,
+                        description = "AWS secret key. This cannot be used along with the credential.provider.class",
+                        optional = true,
+                        defaultValue = " "
+                ),
+                @Parameter(
+                        name = "bucket.name",
+                        type = DataType.STRING,
+                        description = "Name of the S3 bucket"
+                ),
+                @Parameter(
+                        name = "aws.region",
+                        type = DataType.STRING,
+                        description = "The region to be used to create the bucket",
+                        optional = true,
+                        defaultValue = " "
+                ),
+                @Parameter(
+                        name = "versioning.enabled",
+                        type = DataType.BOOL,
+                        description = "Flag to enable versioning support in the bucket",
+                        optional = true,
+                        defaultValue = "false"
+                ),
+                @Parameter(
+                        name = "object.path",
+                        type = DataType.STRING,
+                        description = "Path for each S3 object",
+                        dynamic = true
+                ),
+                @Parameter(
+                        name = "storage.class",
+                        type = DataType.STRING,
+                        description = "AWS storage class",
+                        optional = true,
+                        defaultValue = "standard"
+                ),
+                @Parameter(
+                        name = "flush.size",
+                        type = DataType.INT,
+                        description = "Maximum number of events to be written into a file",
+                        optional = true,
+                        defaultValue = "1"
+                ),
+//                @Parameter(
```

Comment:
Are we commenting here to support these later?

Reply:
Yes. But removed in the latest commit.